File: src/main/java/io/github/saneea/dvh/feature/clipboard/TextToClipboard.java
package io.github.saneea.dvh.feature.clipboard;

import java.awt.Toolkit;
import java.awt.datatransfer.Clipboard;
import java.awt.datatransfer.StringSelection;

import io.github.saneea.dvh.Feature;
import io.github.saneea.dvh.FeatureContext;

public class TextToClipboard implements Feature, Feature.In.Text.String {

	private String in;

	@Override
	public Meta meta(FeatureContext context) {
		return Meta.from("write text to clipboard");
	}

	@Override
	public void run(FeatureContext context) throws Exception {

		Clipboard clipboard = Toolkit.getDefaultToolkit().getSystemClipboard();

		StringSelection selection = new StringSelection(in);
		clipboard.setContents(selection, selection);
	}

	@Override
	public void setIn(String in) {
		this.in = in;
	}

}

File: src/main/java/io/github/saneea/dvh/feature/clipboard/TextToClipboard.kt
package io.github.saneea.dvh.feature.clipboard

import io.github.saneea.dvh.Feature
import io.github.saneea.dvh.Feature.Meta
import java.awt.Toolkit
import java.awt.datatransfer.StringSelection

class TextToClipboard :
    Feature,
    Feature.In.Text.String {

    override lateinit var inTextString: String

    override val meta = Meta("write text to clipboard")

    override fun run() {
        val clipboard = Toolkit.getDefaultToolkit().systemClipboard
        val selection = StringSelection(inTextString)
        clipboard.setContents(selection, selection)
    }
}